### PR TITLE
test: add unit tests for tab broadcast utilities

### PR DESCRIPTION
**Pull request by cuspymd**
🎯 **What:**
Added unit tests for `shared/tab-broadcast.js`.

📊 **Coverage:**
- `broadcastToAllTabs`: verified message sending to all tabs and error handling.
- `broadcastToTabsByUrl`: verified URL filtering, skipping invalid tabs, and error handling.

✨ **Result:**
Improved reliability of tab messaging utilities by ensuring they handle common edge cases (like missing content scripts) gracefully. Verified that `browserAPI` correctly integrates with the global `chrome` mock in the test environment.

---
*PR created automatically by Jules for task [6443564229914185812](https://jules.google.com/task/6443564229914185812) started by @cuspymd*

### CODE REVIEW

File: tests/tab-broadcast.test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    it('should handle errors gracefully', async () => {
+      const url = 'https://example.com/*';
+      const tabs = [{ id: 1 }];
+      chrome.tabs.query.mockResolvedValue(tabs);
+      const message = { action: 'test' };
+
+      chrome.tabs.sendMessage.mockRejectedValue(new Error('Failed'));
+
+      await expect(broadcastToTabsByUrl(url, message)).resolves.not.toThrow();
```
[the remark this answers is not in the document]
Comment (cuspymd) [reply tests/tab-broadcast.test.js]: > **P1 Badge Assert async success without `toThrow` on resolved value**
> 
> This assertion uses `resolves.not.toThrow()` on a promise returned by `broadcastToTabsByUrl`, but in Jest `toThrow` only works when the received value is a function; after `resolves`, the received value is the resolved result (`undefined`), which makes the matcher itself fail. In environments that run this test, the case will fail regardless of the function’s actual error handling behavior, causing unnecessary CI/test breakage.
> 
> Useful? React with 👍 / 👎.

@jules
